src/common/runnable/run-many-sync-for.ts

```diff
@@ -16,33 +15,34 @@ export interface RunnableSync<TParameter = void> {
 
 type RunSync<Param> = (parameter: Param) => void;
 
-export type RunManySync = <Param>(
-  injectionToken: InjectionToken<Runnable<Param>, void>
-) => RunSync<Param>;
+export type RunManySync = <Param>(injectionToken: InjectionToken<RunnableSync<Param>, void>) => RunSync<Param>;
+
+function runCompositeRunnableSyncs<Param>(param: Param, composite: Composite<RunnableSync<Param>>) {
```

Comment:
Hmm. The same thing could be achieved without recursion using `getCompositeNormalization`.